cpp/ql/test/library-tests/dataflow/taint-tests/movableclass.cpp

```diff
@@ -19,10 +19,10 @@ class MyMovableClass {
 	int v;
 };
 
-MyMovableClass &&getUnTainted() { return MyMovableClass(1); }
-MyMovableClass &&getTainted() { return MyMovableClass(source()); }
```

Comment:
Were these (`&&`) just not necessary to get move semantics all the way through lines 59 and 60?

Reply:
It's not so much that they weren't necessary, as that they weren't well-defined behavior. If you return a reference to a temporary by rvalue reference, it's still a dangling reference. To return a movable object, you just return it by value. The construction of the return value will use the move constructor (although in this case, the constructor is probably elided entirely). When you then use the return value in the caller, that return value is an xvalue, so it will get move semantics as well.